src/main/java/com/ebivariation/contigalias/service/AssemblyService.java

```diff
@@ -52,54 +54,68 @@ public AssemblyService(AssemblyRepository repository, @Qualifier("NCBIDataSource
+        } else return new LinkedList<>();
     }
 
-    public Optional<AssemblyEntity> getAssemblyByGenbank(String genbank) {
+    public List<AssemblyEntity> getAssemblyByGenbank(String genbank) {
         Optional<AssemblyEntity> entity = repository.findAssemblyEntityByGenbank(genbank);
-        entity.ifPresent(this::stripAssemblyFromChromosomes);
-        return entity;
+        return convertOptionalToList(entity);
     }
 
-    public Optional<AssemblyEntity> getAssemblyByRefseq(String refseq) {
+    public List<AssemblyEntity> getAssemblyByRefseq(String refseq) {
         Optional<AssemblyEntity> entity = repository.findAssemblyEntityByRefseq(refseq);
-        entity.ifPresent(this::stripAssemblyFromChromosomes);
-        return entity;
+        return convertOptionalToList(entity);
     }
 
-    public List<AssemblyEntity> getAssembliesByTaxid(long taxid) {
-        List<AssemblyEntity> entityList = repository.findAssemblyEntitiesByTaxid(taxid);
-        if (!entityList.isEmpty()) {
-            entityList.forEach(this::stripAssemblyFromChromosomes);
-        }
-        return entityList;
+    public List<AssemblyEntity> getAssembliesByTaxid(long taxid, Pageable request) {
+        Slice<AssemblyEntity> slice = repository.findAssemblyEntitiesByTaxid(taxid, request);
+        return convertSliceToList(slice);
     }
 
-    public Optional<AssemblyEntity> fetchAndInsertAssembly(
-            String accession) throws IOException, IllegalArgumentException {
+    public void fetchAndInsertAssembly(String accession)
+            throws IOException, IllegalArgumentException {
         Optional<AssemblyEntity> entity = repository.findAssemblyEntityByAccession(accession);
         if (entity.isPresent()) {
             throw duplicateAssemblyInsertionException(accession, entity.get());
         }
         Optional<AssemblyEntity> fetchAssembly = dataSource.getAssemblyByAccession(accession);
         fetchAssembly.ifPresent(this::insertAssembly);
-        return fetchAssembly;
     }
 
-    public Optional<AssemblyEntity> getAssemblyByAccession(String accession) {
-        Optional<AssemblyEntity> assembly = repository.findAssemblyEntityByAccession(accession);
-        assembly.ifPresent(this::stripAssemblyFromChromosomes);
-        return assembly;
+    public List<AssemblyEntity> getAssemblyByAccession(String accession) {
+        Optional<AssemblyEntity> entity = repository.findAssemblyEntityByAccession(accession);
+        return convertOptionalToList(entity);
+    }
+
+    public List<AssemblyEntity> convertSliceToList(Slice<AssemblyEntity> slice) {
+        if (slice.getNumberOfElements() > 0) {
+            List<AssemblyEntity> content = slice.getContent();
+            content.forEach(this::stripAssemblyFromChromosomes);
+            return content;
+        } else {
+            return new LinkedList<>();
+        }
+    }
+
+    public List<AssemblyEntity> convertOptionalToList(Optional<AssemblyEntity> optional) {
```

Comment:
probably this method should not exist. if we are converting an optional into a list we might be doing something wrong.

Reply:
mmm, ok if we want a uniform REST interface, we might need this method in the controller.